### PR TITLE
Implement trait for arrays and tuples

### PR DESCRIPTION
**Pull request by nearmax**

### CODE REVIEW

**Comment by bowenwang1996** [comment]
@ailisp can you set up CI for this repo?

**Comment by ailisp** [comment]
Sure. I’m waiting for our github organization admin confirm a Travis
subscription. Travis seems to have best rust support so far, with one line
in config it caches .cargo automatically. Other popular rust project like
parity use it too

File: borsh-rs/borsh/src/de/mod.rs [hunk 1]
```diff
@@ -261,3 +248,58 @@ impl BorshDeserialize for Box<[u8]> {
...
+      {
+        #[inline]
+        fn deserialize<R: Read>(reader: &mut R) -> Result<Self, Error> {
+            Ok(($($name::deserialize(reader)?,)+))
+        }
+      }
+    };
+}
+
+//impl_tuple!(T0);
```
[the remark this answers is not in the document]
Comment (nearmax) [reply borsh-rs/borsh/src/de/mod.rs]: Leaked debug code. `(T)` and `T` are the same types.

File: borsh-rs/borsh/src/de/mod.rs [hunk 2]
```diff
@@ -261,3 +248,58 @@ impl BorshDeserialize for Box<[u8]> {
...
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12 T13);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12 T13 T14);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12 T13 T14 T15);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12 T13 T14 T15 T16);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12 T13 T14 T15 T16 T17);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12 T13 T14 T15 T16 T17 T18);
+impl_tuple!(T0 T1 T2 T3 T4 T5 T6 T7 T8 T9 T10 T11 T12 T13 T14 T15 T16 T17 T18 T19);
```
[the remark this answers is not in the document]
Comment (nearmax) [reply borsh-rs/borsh/src/de/mod.rs]: Such macro would be too complex to maintain.

**Comment by nearmax** [comment]
PTAL

**Pull request merged**